app/code/Magento/SalesRule/view/frontend/web/template/payment/discount.html

```diff
@@ -22,12 +22,23 @@
                         <span data-bind="i18n: 'Enter discount code'"></span>
                     </label>
                     <div class="control">
+                        <!-- ko ifnot: isApplied() -->
```

Comment:
This whole condition may be simplified to something like `data-bind="attr:{disabled:isApplied()}"`.
This will allow to avoid copy-pasting the code